src/components/ModalBody/MovieModal.tsx

```diff
@@ -1,54 +1,194 @@
 
-export const MovieModal:React.FunctionComponent = () => {
+export const MovieModal:React.FunctionComponent<{ modalType: string }> = ({ modalType }) => {
     const dispatch = useDispatch()
-    const resetHandler = () => {
-        toggleEditModal()
-        toggleAddModal()
+    const selectedMovieId = useSelector((state: RootState) => state.appData.movieId)
+    const closeModal = () => {
+        if (modalType === "isAddModalOpened") {
+            dispatch(toggleModal("isAddModalOpened", false))
+        } else {
+            dispatch(toggleModal("isEditModalOpened", false))
+        }
+    }
+
+    const initialValues = {
+        id: undefined,
+        title: "",
+        poster_path: "",
+        release_date: "",
+        vote_average: "",
+        runtime: "",
+        overview: "",
+        genres: []
     }
-    const toggleEditModal = () => dispatch(
-        toggleModal("isEditModalOpened", true)
-    )
-    const toggleAddModal = () => dispatch(
-        toggleModal("isAddModalOpened", true)
-    )
 
     return (
         <div className="movie-modal">
-            <div className="movie-modal__top-fields">
-                <div className="movie-modal__top-col-wide">
-                    <LabeledInput type="text" label="Title" placeholder="Movie Title" name="movie_title" id="edit-movie-title" />
-                    <LabeledInput type="url" label="Movie url" placeholder="https://" name="movie_url" id="edit-movie-url" />
-                    <CustomSelect id="edit-movie-genre" label="Genre" placeholder="Select genre" theme="as-field" >
-                        <ul>
-                            <li>
-                                <CustomCheckbox />
-                            </li>
-                            <li>
-                                <CustomCheckbox />
-                            </li>
-                        </ul>
-                    </CustomSelect>
-                </div>
-                <div className="movie-modal__top-col-narrow">
-                    <LabeledInput type="date" label="Release Date" name="movie_date" id="edit-movie-date" />
-                    <LabeledInput type="number" label="Rating" placeholder="7.8" name="movie_rating" id="edit-movie-rating" />
-                    <LabeledInput type="text" label="Runtime" placeholder="minutes" name="movie_runtime" id="edit-movie-runtime" />
-                </div>
-            </div>
-            <div>
-                <LabeledInput type="textarea" label="Overview" placeholder="Movie description" name="movie_overview" id="edit-movie-overview" />
-            </div>
-            <div className="modal__footer">
-                <Button clickHandler={ resetHandler } title="Reset" type="info-button" />
-                <Button clickHandler={ resetHandler } title="Submit" type="primary-button" />
-            </div>
+            <Formik
+                initialValues={ initialValues }
+                validate={ values => {
```

Comment:
Validation logic and submit handlers don't fit into component body. To be able to re-use it and keep your component's code cleaner, you may abstract that away.